File: veebipood/src/main/java/ee/gert/veebipood/Product.java
package ee.gert.veebipood;
// jakarta is hibernate
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
@Getter
@Setter
@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    String name;
    double price;
    String image;
    boolean active;
    String description;

    public Product(String name) {
        this.name = name;
    }

}

File: veebipood/angular-frontend/src/app/Models/Product.ts
import { Nutrients } from "./Nutrients"

export class Product{
    constructor(
        public id : number,
        public name : string,
        public price : number,
        public image : string,
        public active : boolean,
        public description : string,
        public nutrients : Nutrients,
        public category : {}
    ){}
}